### PR TITLE
Add a model for multinomial logistic regression

### PR DESCRIPTION
**Pull request by pobonomo**

### CODE REVIEW

**Comment by bzfwunde** [comment]
Here are some suggested changes:
- Rename the `output_type` parameter to `prediction_method` with the following options, that mostly match the sklearn predictor method names:
  - `predict`
  - `predict_proba`
  - `decision_function`
  - `predict_onehot` (return a 0-1 vector)
- Change the `get_error` method to always work on the probabilities.  For 0/1 return values, the error would also just be 0 or 1 which is not very informative

**Comment by pobonomo** [comment]
> * Rename the `output_type` parameter to `prediction_method` with the following options, that mostly match the sklearn predictor method names:
>   
>   * `predict`
>   * `predict_proba`
>   * `decision_function`
>   * `predict_onehot` (return a 0-1 vector)
> * Change the `get_error` method to always work on the probabilities.  For 0/1 return values, the error would also just be 0 or 1 which is not very informative

This is more involved than I thought.

Doing it for the logistic regression is easy but it has to also work for the pipeline.

File: src/gurobi_ml/sklearn/logistic_regression.py [hunk 1]
```diff
@@ -223,6 +235,12 @@ def _two_classes_model(self, **kwargs):
                 gen_constr.setAttr(attr, val)
         self.gp_model.update()
 
+        if self.predict_function == "predict_proba" and self.predict_class != 1:
+            self.gp_model.addConstr(log_result[:, 0] == self.output[:, 1])
+            self.gp_model.addConstr(self.output[:, 0] == 1 - self.output[:, 1])
+
+        self.gp_model.write("debug.lp")
```
[the remark this answers is not in the document]
Comment (bzfwunde) [reply src/gurobi_ml/sklearn/logistic_regression.py:242]: But I don't understand the `predict_class` parameter.

**Comment by pobonomo** [comment]
@bzfwunde I removed this predict_class parameter as we had discussed and reworked the Janos example.

It has to do some compromises for the example but it's the best I could think off.

**Pull request closed without merging**